src/adagrams.js

```diff
@@ -1,15 +1,183 @@
+    }, {});
+    for (const [letter, count] of Object.entries(letterCountGuessedLetters)) {
+        //checks if the letter is in the hand
+        if (count <= letterCountLettersInHand[letter]) {
+            //if the letter is in the hand, true or false will be added to the boolean list
+            booleanList.push(true);
+        } else {
+            booleanList.push(false);
+        }
+    }
+    if (booleanList.every(Boolean)) {
+        //if all the booleans are true, return true
+        return true;
+    } else {
+        return false;
+    }
 };
 
 export const scoreWord = (word) => {
-  // Implement this method for wave 3
+    const scoreChart = {
+        1: ['A', 'E', 'I', 'O', 'U', 'L', 'N', 'R', 'S', 'T'],
+        2: ['D', 'G'],
+        3: ['B', 'C', 'M', 'P'],
+        4: ['F', 'H', 'V', 'W', 'Y'],
+        5: ['K'],
+        8: ['J', 'X'],
+        10: ['Q', 'Z'],
+    };
+    word = word.toUpperCase();
+    let score = 0;
+    if (word === '') {
+        //if word is empty, return 0
+        return score;
+    }
+    for (const letter of word) {
+        //for each letter in word
+        for (const [points, letters] of Object.entries(scoreChart)) {
+            //for each letter in scoreChart
+            if (letters.includes(letter)) {
+                //if letter is in scoreChart
+                score += parseInt(points); //add points to score
+            }
+        }
+    }
+    if (word.length >= 7) {
+        score += 8;
+    }
+    return score;
 };
 
 export const highestScoreFrom = (words) => {
-  // Implement this method for wave 1
-};
+    const scores = {};
+    for (const word of words) {
+        let scoredWord = scoreWord(word);
+        scores[word] = scoredWord;
+    }
+    const getMaxScore = (object) => {
+        return Object.keys(object).filter((x) => {
+            return object[x] == Math.max.apply(null, Object.values(object)); // get the max value
+        });
+    };
+    const highestScoringWords = getMaxScore(scores);
+    if (highestScoringWords.length > 1) {
+        // get longest word
+        const longestWord = highestScoringWords.reduce(function(a, b) {
+            return a.length > b.length ? a : b;
+        });
+        // if the longest word is 10 characters long, return it - if there are more than one, return the first one
+        if (longestWord.length === 10) {
+            let tenLongWordCount = 0;
+            const tenLongWords = [];
+            for (const word of highestScoringWords) {
+                if (word.length === 10) {
+                    tenLongWordCount += 1;
+                    tenLongWords.push(word);
+                }
+            }
+            if (tenLongWordCount > 1) {
+                // if there are more than one 10-letter words, return the first one
+                return { word: tenLongWords[0], score: scoreWord(tenLongWords[0]) };
+            } else {
+                return { word: longestWord, score: scoreWord(longestWord) }; //
+            }
```

Comment:
If there is a 10 letter word in the input, if we update the comparison in: 
```
const longestWord = highestScoringWords.reduce(function(a, b) {
    return a.length > b.length ? a : b;
});
```
to use `>=` rather than `>`,  `longestWord` will be holding the first 10-letter word in the highest scores. If we did that, we could remove most of the code inside the if-statement `if (longestWord.length === 10) {` and immediately return `longestWord` and its score.